app/uk/gov/hmrc/personaldetailsvalidation/model/PersonalDetails.scala

```diff
@@ -18,10 +18,68 @@ package uk.gov.hmrc.personaldetailsvalidation.model
 
 import java.time.LocalDate
 
+import play.api.libs.json._
 import uk.gov.hmrc.domain.Nino
 import uk.gov.hmrc.model.NonEmptyString
 
-case class PersonalDetails(firstName: NonEmptyString,
+trait PersonalDetails {
+  def toJson: JsObject
```

Comment:
I don't feel we should be having `toJson` on domain objects. 
You can possibly create extension methods using implicits.

Reply:
I disagree, we should not be adding layers of abstraction just for the sake of adding layers of abstraction.

IMO you should not create extension methods, except for objects that you do not own.

Reply:
One of the aspects of functional programming is to keep functions separated from data. We usually try to follow that whenever we can. It is a matter of preference.
Also, even from OOP aspect, just because a class has data, it doesn't mean we should be creating any method on that class. Usually, domain classes should have domain related methods.
I leave it upto you how you want to do it.